src/ai/backend/manager/services/metric/container_metric.py

```diff
@@ -80,14 +78,20 @@ class MetricResponse(BaseModel):
 
 
 class ContainerUtilizationMetricService:
-    _metric_query_endpoint: yarl.URL
-    _range_vector_timewindow: str  # 1m by default
+    _config_provider: ManagerConfigProvider
 
     def __init__(self, config_provider: ManagerConfigProvider) -> None:
-        metric_config = config_provider.config.metric
-        metric_query_addr: HostPortPair = metric_config.address.to_legacy()
-        self._metric_query_endpoint = yarl.URL(f"http://{metric_query_addr}/api/v1")
-        self._range_vector_timewindow = cast(str, metric_config.timewindow)
+        self._config_provider = config_provider
+
+    @property
+    def _metric_query_endpoint(self) -> yarl.URL:
+        metric_query_addr = self._config_provider.config.metric.address.to_legacy()
+        return yarl.URL(f"http://{metric_query_addr}/api/v1")
```

Comment:
We should have received whether it was https or http from the config, that's a bit of a shame.
ex: `address: "https://localhost:9090"`